qiskit/algorithms/evolvers/pvqd.py

```diff
@@ -0,0 +1,472 @@
+                """Evaluate the gradient with the parameter-shift rule.
+
+                This is hardcoded here since the gradient framework does not support computing
+                gradients for overlaps.
+
+                Args:
+                    displacement: The parameters for the ansatz.
+
+                Returns:
+                    The gradient.
+                """
+                # construct lists where each element is shifted by plus (or minus) pi/2
+                dim = displacement.size
+                plus_shifts = (displacement + np.pi / 2 * np.identity(dim)).tolist()
+                minus_shifts = (displacement - np.pi / 2 * np.identity(dim)).tolist()
+
+                evaluated = evaluate_loss(plus_shifts + minus_shifts)
+
+                gradient = (evaluated[:dim] - evaluated[dim:]) / 2
+
+                return gradient
+
+        else:
+            evaluate_gradient = None
+
+        return evaluate_loss, evaluate_gradient
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        """
+        Args:
+            evolution_problem: The evolution problem containing the hamiltonian, total evolution
+                time and observables to evaluate. Note that :class:`~.PVQD` currently does not support
+                hamiltonians of type :class:`~.MatrixOp`.
+
+        Returns:
+            A result object containing the evolution information and evaluated observables.
+
+        Raises:
+            ValueError: If the evolution time is not positive or the timestep is too small.
+        """
+        self._validate_setup()
+
+        time = evolution_problem.time
+        observables = evolution_problem.aux_operators
+        hamiltonian = evolution_problem.hamiltonian
+
+        if not 0 < self.timestep <= time:
+            raise ValueError(
+                f"The time step ({self.timestep}) must be larger than 0 and smaller equal "
+                f"the evolution time ({time})."
+            )
+
+        # get the function to evaluate the observables for a given set of ansatz parameters
+        if observables is not None:
+            evaluate_observables = _get_observable_evaluator(
+                self.ansatz, observables, self.expectation, self._sampler
+            )
+            observable_values = [evaluate_observables(self.initial_parameters)]
+
+        fidelities = [1]
+        times = [0]
+        parameters = [self.initial_parameters]
+
+        current_time = 0
+        initial_guess = self.initial_guess
+
+        while current_time < time:
+            # perform VQE to find the next parameters
+            next_parameters, fidelity = self.step(
+                hamiltonian, parameters[-1], self.timestep, initial_guess
+            )
+
+            # set initial guess to last parameter update
+            initial_guess = next_parameters - parameters[-1]
+
+            # store parameters
+            parameters.append(next_parameters)
```

Comment:
```suggestion
            parameters.append(next_parameters)
```

It is rather self-explanatory.